src/backend/utils/basic_analysis.js

```diff
@@ -71,17 +71,17 @@ module.exports = function analyzeText(text_) {
 
   // asynchonize function
   // to get a basic analysis information, getAsyAnalysis should be called
-  this.getAsyAnalysis = async function() {
-    return new Promise((res, rej) => {
-      if (isValidString) {
-        res(analysis);
-      } else {
-        res(textInfo);
-      }
-      if (isValidString === undefined) {
-        rej(textInfo);
-      }
-    });
+  this.getAsyAnalysis = function() {
+    let result;
```

Comment:
Drop this `result` arg, and just use `return` on every line where you set `result` below.  [Return early, return often](https://blog.humphd.org/vocamus-1421/).